src/NetworkInterface.cpp

```diff
@@ -12082,14 +12082,185 @@ void NetworkInterface::getHostsByPort(lua_State *vm) {
     walker(&begin_slot, true , walker_hosts,
             get_udp_hosts_by_port, &count);
 
-  sort_hosts_details(vm, &count, protocol);
+  sort_hosts_details(vm, &count, protocol, false);
 
 }
 
+/* **************************************************** */
+
+bool NetworkInterface::get_hosts_by_udp_service(GenericHashEntry *node, 
```

Comment:
The methods for TCP and UDP are almost identical: please merge them instead of duplicating code